src/components/Input/index.js

```diff
@@ -9,43 +9,37 @@ import StyledInput from './styles/StyledInput'
  * An input field.
  *
  */
-class Input extends PureComponent {
-  render() {
-    const {
-      className,
-      id,
-      form,
-      name,
-      onChange,
-      forwardedRef,
-      pattern,
-      placeholder,
-      size,
-      type,
-      value,
-      variance,
-      ...rest
-    } = this.props
-
-    return (
-      <StyledInput
-        className={className}
-        form={form}
-        id={id}
-        name={name}
-        onChange={onChange}
-        pattern={pattern}
-        placeholder={placeholder}
-        size={size}
-        type={type}
-        value={value}
-        ref={forwardedRef}
-        variance={variance}
-        {...rest}
-      />
-    )
-  }
-}
+const Input = ({
+  className,
+  id,
+  form,
+  name,
+  onChange,
+  forwardedRef,
+  pattern,
+  placeholder,
+  size,
+  type,
+  value,
+  variance,
+  ...rest
+}) => (
+  <StyledInput
+    className={className}
+    form={form}
+    id={id}
+    name={name}
+    onChange={onChange}
+    pattern={pattern}
+    placeholder={placeholder}
+    size={size}
+    type={type}
+    value={value}
+    ref={forwardedRef}
+    variance={variance}
+    {...rest}
```

Comment:
I think here the use of  `React.memo()` would be justified because of the `{...rest}` prop passed

Reply:
Especially because of the `rest` prop passed, I don't think `memo` is a good idea here, because you will try to optimize the component without knowing if in `rest` there is a prop changing at each render. The component should be optimized, only if needed, in the user code, not here.
What do you think ?

Reply:
I agree, you got a point. I'll resolve the comments about this